client/src/components/menubar/index.tsx

```diff
@@ -111,6 +112,23 @@ class MenuBar extends React.PureComponent<{}, State> {
     };
   }
 
+  componentDidUpdate(prevProps: any): void {
+    // @ts-expect-error ts-migrate(2339) FIXME: Property 'dispatch' does not exist on type 'Readon... Remove this comment to see the full error message
+    const { layoutChoice, dispatch } = this.props;
+    const prevConditionMet =
+      prevProps.layoutChoice &&
+      prevProps.layoutChoice.current?.includes(globals.spatialEmbeddingKeyword);
```

Comment:
why not just:
prevProps.layoutChoice?.current?.includes(globals.spatialEmbeddingKeyword);